core/src/requests/request_id.rs

```diff
@@ -138,11 +138,7 @@ impl<T: RequestIdType> fmt::Debug for RequestId<T> {
 
 impl<T> Clone for RequestId<T> {
     fn clone(&self) -> Self {
-        Self {
-            locator: self.locator,
-            counter: self.counter,
-            _phantom_request_type: self._phantom_request_type.clone(),
-        }
+        *self
```

Comment:
No, I don't think so, because generic, and phantom type, and for same reason other traits like `Eq`, `ParitalEq`, `Ord` are also manually implemented.